Week3/homework/README_Homework.md

```diff
@@ -0,0 +1,78 @@
+# Databases Week3 Homework - Ammar Vehbi Merakli
+
+## Raw body to be used in postman as JSON (application/json):
+
+```
+{
+"user": "xxx",
+"email": "xxx@gmail.com",
+"todoListName": "xxx",
+"reminder": "dd/mm/yyyy",
+"todo": "xxx",
+"done": "false",
+"dueDate": "dd/mm/yyyy",
+"tag": "xxx"
+}
+```
+
+## Follow the steps to use functions properly:
+
+1 - Use the dump file as source from mysql prompt to make your entire database ready to be used:
+`source path/to/the/dump/file/ToDoApp_dump.sql`
+
+2 - Run the application with the command `node index.js`
+
+3 - You should run the functions by
+
+    - selecting proper method
+
+    - using proper url
+
+    - inputting necessary information if needed at body in postman
+
+## Functions:
+
+**Users Part:**
+
+    - To create a new user (POST):
+    `http://localhost:3001/ToDoApp/createNewUser`
+
+    - To update a user (PUT):
+    `http://localhost:3001/ToDoApp/updateUser/:userId`
+
+    - To get all users (GET):
+    `http://localhost:3001/ToDoApp/getAllUsers`
+
+    To remove a user (DELETE):
+    `http://localhost:3001/ToDoApp/removeUser/:userId`
+
+**Todo Lists Part:**
+
+    - To create a new todo list (POST):
+    `http://localhost:3001/ToDoApp/:userId/createNewTodoList`
```

Comment:
**Non-blocking comment:** 

please have a look at the naming of URL in REST interfaces:
https://apiguide.readthedocs.io/en/latest/build_and_publish/use_RESTful_urls.html

Most probably it's something not quite clear at first glance, so I would be happy to explain it to you tomorrow or in Slack if you want.

**I'm saying so because** you'll need it for your interviews, and its gonna be important for other team members

Reply:
I took a look in the link you suggested and tried to update my code accordingly. :pray: